WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/VerticalsSearchResultsProvider.kt

```diff
@@ -0,0 +1,17 @@
+package org.wordpress.android.ui.sitecreation.verticals
+
+import dagger.Reusable
+import org.wordpress.android.ui.sitecreation.verticals.SiteCreationIntentsViewModel.IntentListItemUiState
+import org.wordpress.android.util.LocaleManagerWrapper
+import javax.inject.Inject
+
+@Reusable
+class VerticalsSearchResultsProvider @Inject constructor(localeManagerWrapper: LocaleManagerWrapper) {
+    private val locale = localeManagerWrapper.getLocale()
+
+    fun search(items: List<IntentListItemUiState>, query: String) = if (query.isEmpty()) items else items.filter {
+        val text = it.verticalText.lowercase(locale)
+        val query = query.lowercase(locale)
+        text.contains(query)
+    }
```

Comment:
I love the simplicity of this. As we discussed, this seems fine considering the size of the pre-bundled set, and this is structured in a way that we can easily enhance it later, if necessary. :+1: